### PR TITLE
Dual parallel example

### PR DESCRIPTION
**Pull request by levelfour**
MNIST data-parallel + model+parallel example. This can be executed on the even number of processes.

### CODE REVIEW

File: chainermn/communicators/_base.py [hunk 1]
```diff
@@ -20,6 +20,9 @@ def rank(self):
     def size(self):
         return self.mpi_comm.size
 
+    def split(self, color, key):
```
Comment (keisukefukuda) [inline chainermn/communicators/_base.py]: Maybe we want an explanation on the parameter 'color'

File: examples/mnist/train_mnist_dual_parallel.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+# coding: utf-8
+
```
Comment (keisukefukuda) [inline examples/mnist/train_mnist_dual_parallel.py]: Could you add a short description of what this example demonstrates?

File: examples/mnist/train_mnist_dual_parallel.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+class MLP1(chainermn.MultiNodeChainList):
+    # Model on worker 1.
+    def __init__(self, comm, n_units, n_out):
+        super(MLP1, self).__init__(comm=comm)
+        self.add_link(MLP1Sub(n_units, n_out), rank_in=0, rank_out=0)
+
+
+def main():
```
Comment (keisukefukuda) [inline examples/mnist/train_mnist_dual_parallel.py]: Does this examples requires that `comm.size % 2 == 0` ?

Comment (levelfour) [reply examples/mnist/train_mnist_dual_parallel.py]: Yes. The number of processes must be even.

Comment (levelfour) [reply examples/mnist/train_mnist_dual_parallel.py]: Some assertion might be needed.

File: chainermn/communicators/_base.py (hunk 1 above)
Comment (levelfour) [reply chainermn/communicators/_base.py]: Added some detailed comments.

File: examples/mnist/train_mnist_dual_parallel.py (hunk 2 above)
Comment (levelfour) [reply examples/mnist/train_mnist_dual_parallel.py]: I added some explanation in README.

**Pull request merged**